qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -765,6 +804,9 @@ def __new__(
         envelope_expr = _amp * sym.Piecewise(
             (_gaussian_ledge, _t <= _sq_t0), (_gaussian_redge, _t >= _sq_t1), (1, True)
         )
+        # To conform with some old tests, the angle part is inserted only when needed.
+        if angle != 0:
+            envelope_expr *= sym.exp(1j * _angle)
```

Comment:
I _suspect_ (but don't know) that the use of `1j` might be causing problems.  The sympy/symengine imaginary constant is `sym{py,engine}.I`, whereas using the Python `complex double` `1j` will trigger a potentially lossy conversion.